core/session.go

```diff
@@ -0,0 +1,154 @@
+package core
+
+import (
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/dicedb/dice/config"
+	"golang.org/x/crypto/bcrypt"
+)
+
+const (
+	DefaultUserName = "default"
+	AuthCmd         = "AUTH"
+
+	SessionStatusPending SessionStatusT = SessionStatusT(0)
+	SessionStatusActive  SessionStatusT = SessionStatusT(1)
+	SessionStatusExpired SessionStatusT = SessionStatusT(2)
+)
+
+var (
+	UserStore *Users = NewUsers()
+)
+
+type (
+	SessionStatusT uint8
+	Session        struct {
+		ID   uint64
+		User *User
+
+		CreatedAt      time.Time
+		LastAccessedAt time.Time
+
+		Status SessionStatusT
+	}
+
+	Users struct {
+		store  map[string]*User
+		stLock *sync.RWMutex
+	}
+
+	User struct {
+		Username          string
+		Passwords         []string
+		IsPasswordEnabled bool
+	}
+)
+
+func NewUsers() (users *Users) {
+	users = &Users{
+		store:  make(map[string]*User),
+		stLock: &sync.RWMutex{},
+	}
+	return
+}
+
+func (users *Users) Get(username string) (user *User, err error) {
+	users.stLock.RLock()
+	defer users.stLock.RUnlock()
+	isPresent := false
+	if user, isPresent = users.store[username]; !isPresent {
+		err = fmt.Errorf("user %s not found", username)
```

Comment:
Reformat this error to how Redis raises it.

Reply:
Given that this is an internal error which will not be returned to the client, does this require changing?